lib/AST/ASTContext.cpp

```diff
@@ -841,7 +841,7 @@ static FuncDecl *findLibraryIntrinsic(const ASTContext &ctx,
 
 /// Check whether the given function is non-generic.
 static bool isNonGenericIntrinsic(FuncDecl *fn, bool allowTypeMembers,
-                                  Type &input,
```

Comment:
Reference to ArrayRef seems suspicious.  Is there a way to rewrite this so ownership of the parameters is the responsibility of the outer context (SmallVectorImpl &) or put the callback here?